nodepool/nodepool.py

```diff
@@ -1483,6 +1483,49 @@ def count_nodes_and_subnodes(provider_name):
 
         allocation_history.grantsDone()
 
+        # sort allocations by label demand by demanded amount
+        # does a weighted round-robin using the node demand as priority
+        # convert these lists so the values are mutable inside
+        node_tracker = map(list, nodes_to_launch)
+
+        # empty out nodes_to_launch since we're going to refill it with new ordering
+        nodes_to_launch = []
+
+        while node_tracker:
+            # sort the tracker based on the number that we need to spawn, descending
+            node_tracker = sorted(node_tracker, key=lambda item: item[1], reverse=True)
```

Comment:
I had to recreate the github repo, but elyezer commented that I should use itemgetter here, and he's right! :)